src/clients/pagination/stream.rs

```diff
@@ -9,7 +9,34 @@ use futures::{future::Future, stream::Stream};
 /// Alias for `futures::stream::Stream<Item = T>`, since async mode is enabled.
 pub type Paginator<'a, T> = Pin<Box<dyn Stream<Item = T> + 'a>>;
 
+pub type RequestFuture<'a, T> = Pin<Box<dyn 'a + Future<Output = ClientResult<Page<T>>>>>;
```

Comment:
I have to say, It's a little headache to understand this function signature 😂

Reply:
I see what you mean, coming up with it wasn't easy either. :grinning: In essence it's mostly the signature of a `Future` that returns a `ClientResult<Page<T>>` with the speciality that it is `Box`ed (and pinned) here.